src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -195,6 +196,16 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                                             Diag(DiagnosticDescriptors.RedundantQualifierInMessage, ma.GetLocation(), method.Identifier.ToString());
                                         }
 
+                                        try
+                                        {
+                                            TemplateHelper.Parse(msg, lm.TemplateList);
+                                        }
+                                        catch (ArgumentException ex)
```

Comment:
What do you think about making it a `TryParse` method, and not using exceptions for control flow?

Reply:
OK